File: scripts/resize-cocktail-images.py
#!/usr/bin/env python3
"""
Resize all cocktail images to 400x400 pixels to reduce file sizes and improve load times.
Usage: python resize-cocktail-images.py
"""

import os
import sys
from PIL import Image, ImageOps
from pathlib import Path

def resize_images():
    # Paths
    script_dir = Path(__file__).parent
    cocktails_dir = script_dir.parent / 'assets' / 'cocktails'
    output_dir = script_dir.parent / 'assets' / 'cocktails-resized'
    
    # Create output directory
    output_dir.mkdir(exist_ok=True)
    
    # Get all image files
    image_extensions = {'.jpg', '.jpeg', '.png', '.webp'}
    image_files = [f for f in cocktails_dir.iterdir() 
                   if f.suffix.lower() in image_extensions]
    
    print(f"Found {len(image_files)} images to resize...")
    
    for i, image_path in enumerate(image_files, 1):
        try:
            # Open and resize image
            with Image.open(image_path) as img:
                # Convert to RGB if necessary (for PNG with transparency)
                if img.mode in ('RGBA', 'LA'):
                    background = Image.new('RGB', img.size, (255, 255, 255))
                    background.paste(img, mask=img.split()[-1] if img.mode == 'RGBA' else None)
                    img = background
                elif img.mode != 'RGB':
                    img = img.convert('RGB')
                
                # Resize with proper aspect ratio handling
                img_resized = ImageOps.fit(img, (400, 400), Image.Resampling.LANCZOS)
                
                # Save with optimized quality
                output_path = output_dir / f"{image_path.stem}.jpg"
                img_resized.save(output_path, 'JPEG', quality=85, optimize=True)
                
                print(f"Resized {i}/{len(image_files)}: {image_path.name}")
                
        except Exception as e:
            print(f"Error processing {image_path.name}: {e}")
    
    print(f"\n✅ All images resized successfully!")
    print(f"Resized images saved to: {output_dir}")
    print("After verifying the resized images, replace the original folder.")

if __name__ == "__main__":
    try:
        resize_images()
    except KeyboardInterrupt:
        print("\n❌ Process interrupted by user")
        sys.exit(1)
    except Exception as e:
        print(f"❌ Error: {e}")
        sys.exit(1)

File: scripts/resize-cocktail-images.js
/* eslint-env node */
const fs = require('fs');
const path = require('path');
const sharp = require('sharp'); // npm install sharp

const cocktailsDir = path.join(__dirname, '../assets/cocktails');
const outputDir = path.join(__dirname, '../assets/cocktails-resized');

// Create output directory if it doesn't exist
if (!fs.existsSync(outputDir)) {
  fs.mkdirSync(outputDir, { recursive: true });
}

async function resizeImages() {
  try {
    const files = fs.readdirSync(cocktailsDir);
    const imageFiles = files.filter(file => 
      file.toLowerCase().endsWith('.jpg') || 
      file.toLowerCase().endsWith('.jpeg') || 
      file.toLowerCase().endsWith('.png')
    );

    console.log(`Found ${imageFiles.length} images to resize...`);

    for (let i = 0; i < imageFiles.length; i++) {
      const file = imageFiles[i];
      const inputPath = path.join(cocktailsDir, file);
      const outputPath = path.join(outputDir, file);

      await sharp(inputPath)
        .resize(400, 400, {
          fit: 'cover',
          position: 'center'
        })
        .jpeg({ quality: 85 })
        .toFile(outputPath);

      console.log(`Resized ${i + 1}/${imageFiles.length}: ${file}`);
    }

    console.log('✅ All images resized successfully!');
    console.log(`Resized images saved to: ${outputDir}`);
    console.log('After verifying the resized images, replace the original folder.');
    
  } catch (error) {
    console.error('Error resizing images:', error);
  }
}

resizeImages();